moveit_ros/planning/planning_pipeline/test/planning_pipeline_tests.cpp

```diff
@@ -0,0 +1,113 @@
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Anthony Baker */
+
+#include <memory>
+#include <string>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include "moveit/utils/robot_model_test_utils.h"
+#include "rclcpp/node_interfaces/node_clock_interface.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tf2_ros/buffer.h"
+
+struct MockPlanningPipelineMiddlewareHandle : public planning_scene_monitor::PlanningPipeline::MiddlewareHandle
+{
+  MOCK_METHOD(void, createDisplayPathPublisher, (const std::string&, bool), (override));
+  MOCK_METHOD(void, createReceivedRequestPublisher, (const std::string&, bool), (override));
+  MOCK_METHOD(void, createContactsPublisher, (const std::string&, bool), (override));
+  MOCK_METHOD(std::string, getContactsTopicName, (), (const, override));
+  MOCK_METHOD(void, resetDisplayPathPublisher, (), (override));
+  MOCK_METHOD(void, resetReceivedRequestPublisher, (), (override));
+  MOCK_METHOD(void, resetContactsPublisher, (), (override));
+  MOCK_METHOD(bool, has_parameter, (const std::string&), (const, override));
+  MOCK_METHOD(void, get_parameter, (const std::string&, const std::string&), (const, override));
+  MOCK_METHOD(void, createPlannerPlugin, (const std::string&, const moveit::core::RobotModelConstPtr&), (override));
+  MOCK_METHOD(void, get_parameter, (const std::string&, const std::string&), (const, override));
+  MOCK_METHOD(const planning_interface::PlannerManagerPtr&, getPlannerManager, (), (const, override));
+  MOCK_METHOD(void, createAdapterPlugins, (const std::vector<std::string>&), (override));
+  MOCK_METHOD(bool, plan,
+              (const planning_scene::PlanningSceneConstPtr&, const planning_interface::MotionPlanRequest&,
+               planning_interface::MotionPlanResponse&, std::vector<std::size_t>&),
+              (override));
+  MOCK_METHOD(void, terminatePlannerPlugin, (), (override));
+  MOCK_METHOD(void, publishReceivedRequest, (const planning_interface::MotionPlanRequest&), (override));
+  MOCK_METHOD(void, publishContactsMarkerArray, (const visualization_msgs::msg::MarkerArray&), (override));
+  MOCK_METHOD(void, publishDisplayPathTrejectory, (const moveit_msgs::msg::DisplayTrajectory&), (override));
+};
+
+TEST(PlanningPipelineTests, GeneratePlan)
```

Comment:
As a general comment, it looks like this PR is growing quite big even without tests. I would consider adding simple tests and mainly concentrate on refactoring the code to DI format and then continue with another test focused PRs (ideally one set of tests for one functionality for easier reviews).